novawallet/Common/Services/ChainRegistry/ChainFilterStrategy.swift

```diff
@@ -0,0 +1,94 @@
+import Foundation
+import Operation_iOS
+
+// swiftlint:disable switch_case_alignment
+enum ChainFilterStrategy {
+    typealias Filter = (DataProviderChange<ChainModel>) -> Bool
+    typealias Transform = (DataProviderChange<ChainModel>, ChainModel) -> DataProviderChange<ChainModel>
+
+    case combined([ChainFilterStrategy])
+    
+    case enabledChains
+    case hasProxy
+    case chainId(ChainModel.Id)
+    case noFilter
+    
+    private var filter: Filter {
+        switch self {
+        case .enabledChains: { change in
+                switch change {
+                case .update where change.item?.syncMode.enabled() == true,
+                     .insert where change.item?.syncMode.enabled() == true,
+                     .delete:
+                    true
+                default:
+                    false
+                }
+            }
+        case .hasProxy: { change in
+            #if F_RELEASE
+                return change.item?.hasProxy == true
+                    && change.item?.isTestnet == false
+            #else
+                return change.item?.hasProxy == true
+            #endif
+        }
+        case let .chainId(chainId): { $0.item?.chainId == chainId }
+        case let .combined(strategies): { change in
+            let resultSet = Set(strategies.map { $0.filter(change) })
+            
+            return !resultSet.contains(false)
+        }
+        case .noFilter: { _ in true }
+        }
+    }
+
+    private var transform: Transform? {
+        if case .enabledChains = self {
```

Comment:
What if ```.enabledChains``` part of ```combined``` filter. I would handle that as well by passing the result of previous filter to the next one. I think we need this when we have, for example, ```.combined([.enabledChains, .hasProxy])```